cs_unittest/TestAll.cs

```diff
@@ -0,0 +1,1014 @@
+                {
+                    vw.Learn(dataLine);
+                }
+                vw.RunMultiPass();
+                VWTestHelper.AssertEqual("train-sets/ref/test_named_csoaa_train.stderr", vw.PerformanceStatistics);
+            }
+
+        }
+        // Skipping test, unable to parse prediction file
+        /* Unable to parse command line: Empty path name is not legal.
+        Parameter name: path */
+        /* Unable to parse command line: lines must not be empty. For multi-line examples use Learn(IEnumerable<string>) overload. */
+
+        [TestMethod]
+        [Description("experience replay version of test 1")]
+        [TestCategory("Command line")]
+        // [DeploymentItem(@"train-sets/0001.dat", @"train-sets")]
+        // [DeploymentItem(@"train-sets/ref/0001-replay.stderr", @"train-sets\ref")]
+
+        public void CommandLine_Test94()
+        {
+
+            using (var vw = new VowpalWabbit("-k -l 20 --initial_t 128000 --power_t 1 -d train-sets/0001.dat -f models/0001.model -c --passes 8 --invariant --ngram 3 --skips 1 --holdout_off --replay_b 100"))
+            {
+                foreach (var dataLine in File.ReadLines("train-sets/0001.dat"))
+                {
+                    vw.Learn(dataLine);
+                }
+                vw.RunMultiPass();
+                VWTestHelper.AssertEqual("train-sets/ref/0001-replay.stderr", vw.PerformanceStatistics);
+            }
+
+        }
+
+        [TestMethod]
+        [Description("named labels at training time (csoaa) with experience replay")]
+        [TestCategory("Command line")]
+        // [DeploymentItem(@"train-sets/test_named_csoaa", @"train-sets")]
+        // [DeploymentItem(@"train-sets/ref/test_named_csoaa_train-replay.stderr", @"train-sets\ref")]
+
+        public void CommandLine_Test95()
+        {
+
+            using (var vw = new VowpalWabbit("--named_labels det,noun,verb --csoaa 3 -d train-sets/test_named_csoaa -k -c --passes 10 --holdout_off -f models/test_named_csoaa.model --replay_c 100"))
+            {
+                foreach (var dataLine in File.ReadLines("train-sets/test_named_csoaa"))
+                {
+                    vw.Learn(dataLine);
+                }
+                vw.RunMultiPass();
+                VWTestHelper.AssertEqual("train-sets/ref/test_named_csoaa_train-replay.stderr", vw.PerformanceStatistics);
+            }
+
+        }
+        /* Unable to parse command line: Empty path name is not legal.
+        Parameter name: path */
+
+        [TestMethod]
+        [Description("")]
+        [TestCategory("Command line")]
+        // [DeploymentItem(@"train-sets/0001.dat", @"train-sets")]
+        // [DeploymentItem(@"train-sets/ref/0097.stderr", @"train-sets\ref")]
+
+        public void CommandLine_Test97()
+        {
+
+            using (var vw = new VowpalWabbit("-d train-sets/0001.dat -f models/0097.model --save_resume"))
```

Comment:
Why do we need to checkin generated code?  It seems undesirable.

-John

On 11/26/2015 11:47 PM, Markus Cozowicz wrote:
>
> In cs_unittest/TestAll.cs 
> <https://github.com/JohnLangford/vowpal_wabbit/pull/863#discussion_r46017140>:
>
> > +            }
> > +
> > +        }
> > +        /* Unable to parse command line: Empty path name is not legal.
> > +        Parameter name: path */
> > +
> > +        [TestMethod]
> > +        [Description("")]
> > +        [TestCategory("Command line")]
> > +        // [DeploymentItem(@"train-sets/0001.dat", @"train-sets")]
> > +        // [DeploymentItem(@"train-sets/ref/0097.stderr", @"train-sets\ref")]
> > +
> > +        public void CommandLine_Test97()
> > +        {
> > +
> > +            using (var vw = new VowpalWabbit("-d train-sets/0001.dat -f models/0097.model --save_resume"))
> It’s generated by cs_testhelp, but I don’t run it at build time as it 
> actually requires the tests to run (otherwise I’m not able to extract 
> all command line arguments, remember the input file discussion?) From: 
> John <notifications@github.com<mailto:notifications@github.com>> 
> Reply-To: JohnLangford/vowpal_wabbit 
> <reply@reply.github.com<mailto:reply@reply.github.com>> Date: 
> Thursday, November 26, 2015 at 13:57 To: JohnLangford/vowpal_wabbit 
> <vowpal_wabbit@noreply.github.com<mailto:vowpal_wabbit@noreply.github.com>> 
> Cc: Markus Cozowicz 
> <marcozo@microsoft.com<mailto:marcozo@microsoft.com>> Subject: Re: 
> [vowpal_wabbit] C# refactoring, memory leak fixes, general 
> goodness,... (#863 
> <https://github.com/JohnLangford/vowpal_wabbit/pull/863>) In 
> cs_unittest/TestAll.cs<#863 (comment) 
> <https://github.com/JohnLangford/vowpal_wabbit/pull/863#discussion_r46002381>>:
> + } + + } + /* Unable to parse command line: Empty path name is not 
> legal. + Parameter name: path */ + + [TestMethod] + [Description("")] 
> + [TestCategory("Command line")] + // 
> [DeploymentItem(@"train-sets/0001.dat", @"train-sets")] + // 
> [DeploymentItem(@"train-sets/ref/0097.stderr", @"train-sets\ref")] + + 
> public void CommandLine_Test97() + { + + using (var vw = new 
> VowpalWabbit("-d train-sets/0001.dat -f models/0097.model --save_resume"))
> Cut and paste seems ... inelegant. — Reply to this email directly or 
> view it on 
> GitHub<https://github.com/JohnLangford/vowpal_wabbit/pull/863/files#r46002381>.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/JohnLangford/vowpal_wabbit/pull/863/files#r46017140>.
>